pkg/cli/dev.go

```diff
@@ -0,0 +1,177 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	cli "github.com/acorn-io/acorn/pkg/cli/builder"
+	"github.com/acorn-io/acorn/pkg/client"
+	"github.com/acorn-io/acorn/pkg/deployargs"
+	"github.com/acorn-io/acorn/pkg/dev"
+	"github.com/acorn-io/acorn/pkg/rulerequest"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func NewDev(c CommandContext) *cobra.Command {
+	cmd := cli.Command(&Dev{out: c.StdOut, client: c.ClientFactory}, cobra.Command{
+		Use:               "dev [flags] IMAGE|DIRECTORY [acorn args]",
+		SilenceUsage:      true,
+		Short:             "Run an app from an image or Acornfile in dev mode or attach a dev session to a currently running app",
+		ValidArgsFunction: newCompletion(c.ClientFactory, imagesCompletion(true)).withSuccessDirective(cobra.ShellCompDirectiveDefault).withShouldCompleteOptions(onlyNumArgs(1)).complete,
+		Example: `
+acorn dev <IMAGE>
+acorn dev .
+acorn dev --name wandering-sound
+acorn dev --name wandering-sound <IMAGE>
```

Comment:
should open a dev session to app named wandering-sound w/ the image provided

Reply:
How will the image provided be used?

Reply:
either it will replace the image of the preexisting app or start an app with that name and image
```
./bin/acorn dev --name test-me test
test-me
INFO[0001] Updating app [test-me] to image [test]  
```

Reply:
Interesting.